### PR TITLE
Avoid crashing when trying to sort an enum that has only one case

### PR DESCRIPTION
**Pull request by Lutzifer**
Problem:

trying to format

```swift
// swiftformat:sort
enum Test {
  case only
}
```

crashes with

```
$ swiftformat test.swift 
Running SwiftFormat...
Reading config file at <Redacted>/.swiftformat
[1]    88603 trace trap  swiftformat test.swift
```

The workaround is obvious if you know the cause (remove the annotation from single-case enums), but I think we should avoid the crash.

### CODE REVIEW

File: Tests/RulesTests+Organization.swift [hunk 1]
```diff
@@ -2962,6 +2962,25 @@ class OrganizationTests: RulesTests {
...
+//        let output = """
+//        // swiftformat:sort
+//        enum FeatureFlags {
+//            case upsellB
+//        }
+//        """
+//
+//        testFormatting(for: input, output, rule: FormatRules.sortDeclarations)
+//    }
+
```
Comment (nicklockwood) [inline Tests/RulesTests+Organization.swift:2983]: Instead of commenting out the test, remove the output parameter, then it will serve as a valid test for the fix.

File: Sources/Rules.swift [hunk 2]
```diff
@@ -6087,18 +6087,21 @@ public struct _FormatRules {
                 })
                 .map { $0.element }
 
-            // Make sure there's at least one newline between each declaration
-            for i in 0 ... (declarations.count - 2) {
```
Comment (nicklockwood) [inline Sources/Rules.swift:6091]: I think a simpler fix is:

```swift
for i in 0 ... max(0, declarations.count - 2) {
```

**Review by nicklockwood — CHANGES_REQUESTED** [review]
Good catch, thanks for the PR!

File: Sources/Rules.swift (hunk 2 above)
Comment (Lutzifer) [reply Sources/Rules.swift:6091]: Tried that, but i+1 is accessed inside the loop and 0 ... 0 enters the loop once

Comment (nicklockwood) [reply Sources/Rules.swift:6091]: Ah, good point. How about:

```swift
for i in 0 ..< max(0, declarations.count - 1) {
```

File: Tests/RulesTests+Organization.swift (hunk 1 above)
Comment (nicklockwood) [reply Tests/RulesTests+Organization.swift:2975]: Thanks! Also, maybe add a test case for an empty enum too, just to be safe?
